sys/net/gnrc/network_layer/sixlowpan/gnrc_sixlowpan.c

```diff
@@ -112,6 +112,22 @@ void gnrc_sixlowpan_multiplex_by_size(gnrc_pktsnip_t *pkt,
     size_t datagram_size = gnrc_pkt_len(pkt->next);
     DEBUG("6lo: iface->sixlo.max_frag_size = %u for interface %i\n",
           netif->sixlo.max_frag_size, netif->pid);
+
+    gnrc_netif_hdr_t *hdr = pkt->data;
+
+    /* L2 header size depends on dst & src address */
+    /* we have to pass this information to the lower layer via value */
+    uint16_t value = netopt_pdu_size_ctx_pack(netif->l2addr_len,
+                                              hdr->dst_l2addr_len,
+                                              hdr->flags);
+
+    if (gnrc_netapi_get(netif->pid, NETOPT_PDU_SIZE, 0,
+                        &value, sizeof(value)) > 0) {
+        netif->sixlo.max_frag_size = value;
```

Comment:
Huh? `value` isn't the `MAX_PDU_SIZE` so it shouldn't be assigned to this. I see some unforeseeable side-effects when this is done. Rather, `netif->sixlo.max_frag_size` should be a fallback for `value` (which should be renamed `pdu_size` or something like that, if `PDU_SIZE` is not implemented.

Reply:
> I had hoped I could get around renaming all instances of `max_frag_size` :sweat_smile:

And I want to avoid hunting bugs that are caused by this semantic change :-P

Reply:
> Isn't this only ever used on TX?

Should be, yes. (and on init but there it is only set)

Reply:
I just have a feeling that I will end up replacing all instances of `max_frag_size` with `frag_size`, but I can use this to check all the instances of it's use.

Reply:
That would also make the code more readable then :-).

Reply:
I think the fragment size depends on the packet. So it should maybe be encoded in `gnrc_sixlowpan_frag_fb_t`. It already has a `hint` field.